src/peep/staff.c

```diff
@@ -268,19 +271,33 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
 
 		RCT2_ADDRESS(RCT2_ADDRESS_STAFF_MODE_ARRAY, uint8)[newStaffId] = STAFF_MODE_WALK;
 
-		for (int edi = 0; edi < 0x80; edi++) {
-			int addr = RCT2_ADDRESS_STAFF_PATROL_AREAS + (newStaffId << 9) + edi * 4;
+		for (int i = 0; i < 0x80; ++i) {
+			int addr = RCT2_ADDRESS_STAFF_PATROL_AREAS + (newStaffId << 9) + i * 4;
 			RCT2_GLOBAL(addr, uint32) = 0;
 		}
 	}
 
-	if(staff_type == STAFF_TYPE_HANDYMAN && gConfigGeneral.handymen_mow_default) {
-		int flags = ((newPeep->staff_orders ^ (1 << 3)) << 8) | 1;
-		game_do_command(newPeep->x, flags, newPeep->y, newPeep->sprite_index, GAME_COMMAND_SET_STAFF_ORDER, (int)newPeep, 0);
+	if ((staff_type == STAFF_TYPE_HANDYMAN) && gConfigGeneral.handymen_mow_default) {
+		uint8 new_orders = newPeep->staff_orders | STAFF_ORDERS_MOWING;
+		game_do_command(newPeep->x, ((int)new_orders << 8) | 1, newPeep->y, newPeep->sprite_index, GAME_COMMAND_SET_STAFF_ORDER, (int)newPeep, 0);
 	}
 
-	*ebx = 0;
-	*edi = newPeep->sprite_index;
+	*newPeep_sprite_index = newPeep->sprite_index;
+	return 0;
+}
+
+/**
+ *
+ *  rct2: 0x006BEFA1
+ */
+void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx, int* esi, int* edi, int* ebp)
+{
+	*ebx = staff_hire_new_staff_member((*ebx & 0xFF00) >> 8,
+									   *ebx & 0xFF,
+									   *eax & 0xFFFF,
+									   *ecx & 0xFFFF,
+									   *edx & 0xFFFF,
+									   edi);
```

Comment:
Github seems to assume tabs are 8 spaces wide, while we use 4 spaces. There doesn't seem to be a way to tell it otherwise, except when viewing whole files. (And even then it doesn't persist.)